### ISSUE TITLE
Another UPnP/`igd` panic

### ISSUE THREAD

**Issue by michaelsproul** [sigp/lighthouse#5020]
## Description

A user on Discord reports a panic when UPnP is enabled:

```
lighthousebeacon  | thread 'tokio-runtime-worker' panicked at 'invalid url or method: Error(InvalidBaseUrl)', /cargo/registry/src/index.crates.io-6f17d22bba15001f/attohttpc-0.16.3/src/request/builder.rs:52:41
lighthousebeacon  | stack backtrace:
lighthousebeacon  |    0: rust_begin_unwind
lighthousebeacon  |              at ./rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/std/src/panicking.rs:593:5
lighthousebeacon  |    1: core::panicking::panic_fmt
lighthousebeacon  |              at ./rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/core/src/panicking.rs:67:14
lighthousebeacon  |    2: core::result::unwrap_failed
lighthousebeacon  |              at ./rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/core/src/result.rs:1651:5
lighthousebeacon  |    3: igd::gateway::Gateway::perform_request
lighthousebeacon  |    4: igd::gateway::Gateway::get_external_ip
lighthousebeacon  |    5: tokio::runtime::task::raw::poll
lighthousebeacon  | note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
lighthousebeacon  | Dec 19 16:32:28.108 CRIT Task panic. This is a bug!              advice: Please check above for a backtrace and notify the developers, message: <none>, task_name: UPnP
lighthousebeacon  | Dec 19 16:32:28.108 INFO Internal shutdown received              reason: Panic (fatal error)
lighthousebeacon  | Dec 19 16:32:28.108 INFO Shutting down..                         reason: Failure("Panic (fatal error)")
lighthousebeacon  | Dec 19 16:32:28.109 INFO Saved DHT state                         service: network
lighthousebeacon  | Dec 19 16:32:28.109 INFO Network service shutdown                service: network
```

## Steps to resolve

Probably requires patching `igd` to make it less panicky (again).

## Version

Lighthouse v4.5.0

## Additional Info

Similar to #4171 which was fixed in v4.2.0 (see https://github.com/sigp/lighthouse/commit/edbb47dd038d1dcfb86f957852a70c7e885270e4).

**Comment by ackintosh** [comment]
Now [igd](https://github.com/sbstp/rust-igd) has been archived, and we are moving from `igd` to [igd-next](https://github.com/dariusc93/rust-igd) in https://github.com/sigp/lighthouse/pull/4840. So I've filed a PR to address the panic issue in igd-next: https://github.com/dariusc93/rust-igd/pull/5

**Comment by ackintosh** [comment]
https://github.com/dariusc93/rust-igd/pull/5 has been merged and released as [igd-next v0.14.3](https://crates.io/crates/igd-next). Therefore, we can update `igd-next` in #4840, or I can make a PR to migrate from `igd` to `igd-next` in order to incorporate the fix into lighthouse.

**Comment by michaelsproul** [comment]
awesome, thank you @ackintosh!

**Comment by ackintosh** [comment]
The fix has been released in [v5.0.0](https://github.com/sigp/lighthouse/releases/tag/v5.0.0) 🎉
